simplify PR instructions

indirectlylit opened the pull request:
# Details

### Summary

The instructions for the PR template were overly complicated:

* they caused people to spend too much time on what ought to be a fast process
* they result in overly complex PRs even for very simple changes

### Reviewer guidance

See if this strikes a better balance between completeness and expediency

# Contributor Checklist

- [x] PR has the correct target milestone
- [x] PR has the appropriate labels
- [ ] Changes in the PR do not introduce accessibility regressions ([confimed by testing with one of the recommended tools](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)) 
- [x] If PR is ready for review, it has been assigned or requests review from someone
- [ ] Documentation is updated as necessary
- [ ] External dependency files are updated (`yarn` and `pip`)
- [ ] If internal dependency is updated, link to diff is included
- [ ] Screenshots of any front-end changes are in the PR description
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] You've added yourself to AUTHORS.rst if you're not there

# Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR has been fully tested manually

indirectlylit commented:
an example of the kind of ugliness this leads to:

![image](https://user-images.githubusercontent.com/2367265/33248664-fad7fcc6-d2da-11e7-9217-f42f54d1d8bd.png)

benjaoming commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -3,16 +3,10 @@
-Tick the item checkbox, ~strikethrough item text~, and write why it was skipped, example:
-
-- [x] ~Skipped item~ This is a documentation fix
+ 4. Feel free to skip checklist items that aren't applicable
```
Shouldn't it say *how* to skip items in the check list since that's the problem we've been having? I don't like the "feel free" part where there's no requirement of making it evident to others if something was skipped...

indirectlylit replied:
In general, I've found the content & context of the PR makes it self-evident when something can be skipped

benjaoming replied:
I don't have a mental recollection of the check list when I review, so if someone just deletes an item, I will just have an odd feeling that something isn't right.

I worry mostly about the deletion of items. But I also worry that skipping them and leaving them unchecked "by design" will create a new practice of just calling shots when merging -- something like "that's probably supposed to be skipped". Once we head down that road, we're going to have problems, because the check lists loose their legitimacy.

indirectlylit replied:
I'd like to find something that allows for both quick and streamlined flows, but also encourages more rigor and process when appropriate.

* Some PRs are literally a single line change with obvious implications
* Some PRs are big, consequential changes that require meticulous review and process

IMO the reviewer and submitter should be allowed some flexibility to move quickly through the process when appropriate, but also demand more info when not appropriate.

Agreed that perhaps deleting sections is not good, because it might make it harder to move from simple to more complex. I'll remove that guidance

benjaoming commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -37,12 +31,12 @@ when applicable, please provide:
 - [ ] External dependency files are updated (`yarn` and `pip`)
 - [ ] If internal dependency is updated, link to diff is included
 - [ ] Screenshots of any front-end changes are in the PR description
+- [ ] PR has been [tested for accessibility regressions](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)
```
This item was intentionally formulated as the above in order for someone to be able to tick it off when it wasn't relevant ;)

benjaoming commented:
Maybe it's better to formulate the check list items like the accessibility item?

> Changes in the PR do not introduce accessibility regressions

This item is stated such that it can be ticked off when it's not relevant...

If people start removing items from the list without making it evident why, we're back to the original problem!

benjaoming commented:
We can also introduce a "soft check list", one that doesn't have tasks? But like a

 - [ ] This PR lives up to [the quality check list](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

Or putting the soft check list in the PR?

### Contributor should make sure that...

 * No accessibility regressions
 * External dependency files are updated
 * If internal dependency is updated, link to diff is included
 * Screenshots of any front-end changes are in the PR description

indirectlylit commented:
made some updates based on feedback:

* removed guidance on deleting sections
* reworded checklist to work better even when they don't apply

also:

* moved instructional text to comments and replaced with '…' placeholder
* included 'PR has been fully tested manually' to Contributor checklist
* changed from first-level to third-level headers because they render smaller in github
